intro-to-R-tidyverse/scripts/00-setup-intro-to-R.R

```diff
@@ -48,46 +48,36 @@ metadata <- readr::read_tsv(file.path(data_dir,
 gene_df <- gene_df %>% 
   dplyr::select(metadata$geo_accession)
 
-# This set up is based on the example 8.7 Factorial Experiment in 
-# https://bioc.ism.ac.jp/packages/2.9/bioc/vignettes/limma/inst/doc/usersguide.pdf
+# This set up is based on the sum to zero parametrization example from 9.5.4 Classic Interaction Models in 
+# https://bioconductor.org/packages/3.1/bioc/vignettes/limma/inst/doc/usersguide.pdf
 
 # Set up combined sex and tissue variable for model
-sex_tissue <- paste0(metadata$tissue, metadata$sex, sep = "_")
-
-# Set variable levels
-sex_tissue <- factor(sex_tissue,
-                     labels = c("control_female", 
-                                "control_male", 
-                                "tumor_female", 
-                                "tumor_male")
-                    )
-
-# Create the design matrix from the genotype information
-des_mat <- model.matrix(~ 0 + sex_tissue)
+sex <- factor(metadata$sex)
+tissue <- factor(metadata$tissue)
+design <- model.matrix(~sex*tissue)
 
 # Neaten up column names
-colnames(des_mat) <- levels(sex_tissue)
+colnames(design) <- c("intercept", "male_female", "astrocytoma_normal", "interaction")
 
 # Apply linear model to data
-fit <- limma::lmFit(gene_df, design = des_mat)
-
-# Set up a contrast matrix that focuses on tumors
-contrast_mat <- limma::makeContrasts(
-  control_v_tumor_female = control_female-tumor_female,
-  control_v_tumor_male = control_male-tumor_male,
-  interaction = (control_female-tumor_female)-(control_male-tumor_male),
-  levels = des_mat)
-
-# Do final fitting
-fit2 <- contrasts.fit(fit, contrast_mat)
+fit <- limma::lmFit(gene_df, design)
 
 # Apply empirical Bayes to smooth standard errors
-fit2 <- eBayes(fit2)
+fit2 <- limma::eBayes(fit)
+
+# collect results for each contrast
+stats_df <- dplyr::bind_rows(
```

Comment:
If we are going to have participants looking at this script at all, we might want to separate out these steps a bit so they are easier to follow. I'll go ahead and separate it out if that is fine with you @jashapiro

Reply:
Not that we are planning on lecturing on this at all, but just that it will be included for their reference, so don't want them to be overwhelmed if they open it.